pkg/provider/github/parse_payload.go

```diff
@@ -173,24 +178,16 @@ func (v *Provider) ParsePayload(ctx context.Context, run *params.Run, request *h
 				if configValueS == "" {
 					continue
 				}
-				split := strings.Split(configValueS, "/")
-				info, _, err := v.Client.Repositories.Get(ctx, split[0], split[1])
-				if err != nil {
-					v.Logger.Warn("we have an invalid repository: `%s` in configmap key or no access to it: %v", configValueS, err)
-					continue
-				}
-				v.repositoryIDs = append(v.repositoryIDs, info.GetID())
+				repoLists = append(repoLists, configValueS)
 			}
 		}
-		var err error
-		if processedEvent.Provider.Token, err = v.GetAppToken(ctx, run.Clients.Kube, event.Provider.URL, installationIDFrompayload, systemNS); err != nil {
+		token, err := v.ScopeGithubTokenToListOfRepos(ctx, repoLists, run, processedEvent)
```

Comment:
the ScopeGithubTokenToListOfRepos can then be private to the github provider struct and reuse from CreateToken